practice-two/src/components/ProductDetails.jsx

```diff
@@ -1,80 +1,120 @@
+ * @param onToggleProductFavorite - Function to handle toggling a product as favorite
+ * @param onClose - Callback function to close the product details.
  * @returns {JSX.Element} Product Details content.
  */
-const ProductDetails = (product = []) => {
-  const selectedProduct = product.product;
+const ProductDetails = ({
+  product = [],
+  favoriteProductIDs = [],
+  onToggleProductFavorite = () => {},
+  onClose = () => {},
+}) => {
+  const selectedProduct = product;
+
   const imageUrls = Object.values(selectedProduct.images);
+
+  const [isPopoverOpen, setIsPopoverOpen] = useState(true);
+
+  const handleButtonClick = () => {
+    setIsPopoverOpen((prevState) => {
+      onClose();
+      return !prevState;
+    });
+  };
+
+  const handleFavoriteToggle = () => {
+    onToggleProductFavorite(product.id);
+  };
+
   return (
-    <section className="container min-h-screen flex justify-center items-center">
-      <div className="box bg-white rounded-2xl shadow-2xl p-10 m-2 w-96 grid grid-cols-1 lg:grid-cols-2 grid-rows-auto md:w-3/4 md:grid-rows-auto items-center">
-        <div className="grid grid-cols-3 grid-rows-auto gap-3 mr-5">
-          {imageUrls.map((imageUrl, index) => (
-            <div
-              // eslint-disable-next-line react/no-array-index-key
-              key={index}
-              className={`border rounded-2xl ${index === 0 ? 'active col-span-3' : ''}`}
-            >
-              <CardImage src={imageUrl} alt={`Product Image ${index + 1}`} />
-            </div>
-          ))}
-        </div>
-        <div>
-          <div className="basic-info flex flex-col gap-4 relative">
-            <div className="flex justify-between">
-              <Typography level={3} className="text-4xl" size="xl">
-                {selectedProduct.name}
-              </Typography>
-              <FavoriteButton />
+    <section
+      className={`fixed inset-0 bg-opacity-70 backdrop-filter backdrop-blur-lg flex items-center justify-center z-50 ${
+        isPopoverOpen ? 'bg-gray-800' : 'hidden'
+      }`}
+    >
+      <div className="container min-h-screen w-[40%] h-[30%] flex justify-center items-center">
+        <Popover
+          closeButtonoo
+          isOpen={isPopoverOpen}
+          onClose={handleButtonClick}
+          popoverClassName="relative bg-white rounded-2xl max-h-[700x] overflow-x-auto shadow-2xl p-10 m-2 "
+          arrowPopover={false}
+        >
+          <div className="grid grid-cols-1 lg:grid-cols-2 grid-rows-none items-center">
+            <div className="grid grid-cols-3 grid-rows-auto gap-3 mr-5">
+              {imageUrls.map((imageUrl, index) => (
+                <div
+                  // eslint-disable-next-line react/no-array-index-key
+                  key={index}
+                  className={`border rounded-2xl ${index === 0 ? 'active col-span-3' : ''}`}
+                >
+                  <CardImage src={imageUrl} alt={`Product Image ${index + 1}`} />
```

Comment:
"Product Image 1", "Product Image 2", "Product Image 3" does this alt make sense??

Reply:
Yes, I udpated into Color '..' styles for phones